### PR TITLE
Charles Emrich built a new Schema, made some routes, and added input validation.

### PR DESCRIPTION
**Pull request by CharlesEmrich**

### CODE REVIEW

**Comment by CharlesEmrich** [comment]
Hey, I just realized that my travis.yml definitely won't be updated to account for the new stuff. I'll get to you about that tomorrow!

File: lib/routes/books.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      // Book.findById(req.params.id).remove()
+      Book.removeById(req.params.id)
+      .then(result => res.send(result))
+      .catch(next);
+  })
+  //Something non-CRUD: Destroy old books.
+  .get('/cull-old', (req, res, next) => {
+    let query = Book.find({pubYear: {$lte: req.body.pubYear}});
+    let promise = query.exec();
+    promise.remove()
```
Comment (goodwid) [inline lib/routes/books.js]: Not sure why you wouldn't just chain these up:
``` js
Book.find({...})
  .exec()
  .remove()
  .then(....
```
vs. storing them in variables and calling methods on them one operation at a time.

File: test/tests.js [hunk 2]
```diff
@@ -5,70 +5,70 @@ const app = require('../lib/server');
...
-    request
-    .delete('/api/2')
-    .end((err, res) => {
-      if (err) return done(err);
-      // console.log(res.text);
-      assert.deepEqual(JSON.parse(res.text), [{'id':2, data: { data: 'barfoobaz'}}]);
-      done();
-    });
-  });
+  //
```
Comment (goodwid) [inline test/tests.js]: Did you intend to leave all your tests commented out?

File: lib/models/book.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  title: {
+    type: String,
+required: true
+  },
+  author: {
+    type: String,
+required: true
+  },
+  pubYear: {
+    type: String || Number, //Does Mongo support Date as a type?
```
Comment (goodwid) [inline lib/models/book.js]: Yes, Date is a supported type.

**Pull request merged**